docs/source/install.rst

```diff
@@ -35,6 +35,8 @@ The current stable version of cocotb requires:
 
 .. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.
 
+.. note:: Type checking cocotb code requires Python 3.11+.
```

Comment:
Maybe add that above to the Python 3.6.2+ line, e.g. 
"At least Python 3.6.2+. We recommend using Python 3.11 or newer to enable type checking."

Reply:
I removed the note. Type checking is often done with a different Python installation or by a program that can parse newer versions of Python that what is running it, so this isn't something that end users need to be concerned about. I left the note in developing.rst, because Sphinx *needs* to read those version-ifdef'd code to generate correct type stubs in the docs.